dev/devicelab/lib/tasks/platform_channels_benchmarks.dart

```diff
@@ -31,7 +31,7 @@ TaskFunction runTask(adb.DeviceOperatingSystem operatingSystem) {
         '.'
       ];
       print('\nExecuting: $flutterExe $createArgs $appDir');
-      utils.eval(flutterExe, createArgs);
+      await utils.eval(flutterExe, createArgs);
```

Comment:
Why was this not caught by the unawaited Futures lint?

Reply:
good question, haven't looked into that yet but I suspect that we're either accidentally ignoring part of the repo for lints or we're ticking a lint bug somehow

Reply:
Looks like this is the cause: https://github.com/flutter/flutter/blob/94f9a2808cbefd24d7340534e0248f9187ad6ece/analysis_options.yaml#L198

Reply:
Pinging @dnfield since that was his PR that disabled that warning.  In this case it would have been helpful since I was using a method I didn't author.  It was easy to assume the call was synchronous.  Is the linter warning not smart enough to catch `.then` usage?  It seems like we'd want to for someone to `await` or have a `.then()`

Another way to have avoided this is to have a linter error if you implicitly discard a return value.  I always liked languages that disallowed that.

Reply:
I'm in the blame but it was for cleanup/bulk moving/revert. It was determined to be WONTFIX here: https://github.com/flutter/flutter/issues/5793

Reply:
We may want to have `dev` enable the lint even if we don't enable it for `packages`.

Reply:
I was hoping that if we turned this on we could just migrate everything to `foobar().then(null)` to explicitly ignore the asynchronous operation.  `then` itself returns a future which means it doesn't eliminate the linter warning.  The linter flag forces you to `await` which isn't always the right case.  This would require people to have to manually ignore like so which isn't bad:
```dart
// ignore: unawaited_futures
utils.eval(flutterExe, createArgs);
```

I think we usually frown upon people having to ignore linter errors to get acceptable behavior although it's fine with me personally.  I'll put in a feature request to the dart language to have a way to explicitly discard an asynchronous operation.

Reply:
@gaaclarke See `unawaited` in the flutter_tools here https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/base/common.dart#L36

Reply:
We could turn it on but add an extension method to futures called `discard()` so that `foobar()` turns into `foobar().discard()`.  It probably wouldn't be much help without the auto fixer knowing about it.

Reply:
https://github.com/dart-lang/sdk/issues/46047

Reply:
The lint doesn't make much sense in Flutter code (a lot of Flutter APIs hand back Futures that are informative but never throw and whose completion does not generally signal a change in API availability and so don't need to be awaited), which is why it's disabled. Enabling it for non-Flutter code (like we have it enabled for the tool) makes sense to me.

Reply:
FWIW in flutter_test we have runtime verification that `await` is being called on appropriate APIs, we could do something similar if this is a specific API that's bug-prone in this way. (Look for the async guard API in flutter_test.)